keps/sig-apps/4026-job-creation-timestamp-annotation/README.md

```diff
@@ -0,0 +1,299 @@
+
+## Design Details
+
+The CronJob controller will only need a minor update to the [getJobFromTemplate2](https://github.com/kubernetes/kubernetes/blob/7024beeeeb1f2e4cde93805a137cd7ad92fec466/pkg/controller/cronjob/utils.go#L188) function, to add the job scheduled timestamp as the job annotation `batch.kubernetes.io/cronjob-scheduled-timestamp`. The scheduled timestamp is represented in `RFC3339`.
+
+For the scheduled timestamp's timezone, the initial thought was to use `UTC` as it's used as the primary one for less confusion. However, since the `job` object has a `spec.timeZone`, it was a better to use the same timezone within the same object. If the job `spec.timeZone` is not set or `nil`, the annotation will use the `UTC` timezone as a default.
+
+### Test Plan
+
+- [X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/controller/cronjob`: `05/22/2023` - `96.2%`
+
+##### Integration tests
+
+Unit tests will ensure the new annotation is correctly added to jobs, and integration tests will verify that the annotation is only added to jobs from newly created CronJobs, not existing workloads.
+
+##### e2e tests
+
+E2E tests will not provide any additional coverage that isn't already covered by unit + integration tests, since we are simply adding an annotation, so no e2e tests will be necessary for this change.
+
+### Graduation Criteria
+
+The feature will be released directly in Beta state since there is no benefit in having an alpha release, since we are simply adding a new annotation so there is very little risk (unlike removing an
+existing annotation which other things may depend on, for example).
+
+#### Beta
+
+- Feature implemented behind the `CronJobsScheduledAnnotation` feature gate.
+- Unit and integration tests passing.
+
+#### GA
+
+Fix any potentially reported bugs.
+
+### Upgrade / Downgrade Strategy
+
+No changes required to existing cluster to use this feature.
+
+### Version Skew Strategy
+
+N/A. This feature doesn't require coordination between control plane components,
+the changes to each controller are self-contained.
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: JobScheduledAnnotation
+  - Components depending on the feature gate:
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+No.
```

Comment:
Actually it does - the job will now contain annotation and it didn't before.

Reply:
Updated.